### DISCUSSION TITLE
How to swap on SPL tokens using Wallet core

### DISCUSSION THREAD

**Discussion by aminsato** [trustwallet/wallet-core#4189] [Q&A] [answered]
How to swap on Solana tokens using Wallet core

**Comment by satoshiotomakan** [comment, marked as answer]
Hi @aminsato, you should integrate a DEX that supports Solana, and then sign the raw transaction bytes using WalletCore.
Something like that
https://github.com/trustwallet/wallet-core/blob/bbb99135847aa71898f085a6c681e3fbb162e530/rust/tw_tests/tests/chains/solana/solana_sign.rs#L885-L925
